main/src/library/File.flix

```diff
@@ -495,7 +532,109 @@ namespace File {
             close(printWriter);
 
             if (checkError(printWriter)) {
-                Err("An error occurred when appending lines to the file: "+f)
+                Err("An error occurred when appending lines to the file: " + f)
+            }
+            else {
+                Result.flatMap((exists) -> Ok(not exists), alreadyExists)
+            }
+        } catch {
+            case ex: ##java.io.IOException =>
+                import java.lang.Throwable.getMessage();
+                Err(getMessage(ex))
+        }
+
+    ///
+    /// Writes `data` to the given `f`.
+    ///
+    /// Creates `f` if it does not exist. Overwrites it if it exists.
+    ///
+    /// Returns `true` if the file `f` was created, and `false` if `f` was overwritten.
+    ///
+    pub def writeBytes(f: String, data: f[Int8]): Result[Bool, String] & Impure with Foldable[f] =
+        try {
+            import new java.io.File(String) as newFile;
+            import new java.io.FileOutputStream(##java.io.File) as newFileStream;
+            import new java.io.PrintWriter(##java.io.Writer) as newPrintWriter;
+            import java.io.FileOutputStream.write(Int32);
+            import java.io.PrintWriter.close();
+            import java.io.PrintWriter.checkError();
+
+            let alreadyExists = File.exists(f);
+
+            let javaFile = newFile(f);
+            let fileStream = newFileStream(javaFile);
+            let printWriter = newPrintWriter(fileStream as ##java.io.Writer);
+
+            Foldable.foldLeft((_, b) -> write(fileStream, Int8.toInt32(b)), (), data);
```

Comment:
`FileOutputStream` has a `write` function to write an array of bytes. It might be cleaner to convert `data` to an array rather than folding over it like this.